### PR TITLE
Enabling cp-ansible on machines with ARM architecture

### PR DESCRIPTION
**Pull request by rrbadiani**
# Description

Allows cp-ansible to run on machines with ARM64 architecture. 
Major changes made -
- When adding apt repository removed the [arch=amd64] to automatically detect the machine architecture
- Modified testing docker image from centos8 to ubi8
- In some scenarios modified the os from debian9 to debian10
- Changed ldap server base machine from centos7 to centos8

EPIC - [ANSIENG-2378](https://confluentinc.atlassian.net/browse/ANSIENG-2378) 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested on jenkins in all 4 cases - 
AMD64 + zookeeper -> [2159](https://jenkins.confluent.io/job/cp-ansible-on-demand/2159/) [2164](https://jenkins.confluent.io/job/cp-ansible-on-demand/2164) [2172](https://jenkins.confluent.io/job/cp-ansible-on-demand/2172/) 
AMD64 + kraft -> [2162](https://jenkins.confluent.io/job/cp-ansible-on-demand/2162/) [2165](https://jenkins.confluent.io/job/cp-ansible-on-demand/2165/) [2171](https://jenkins.confluent.io/job/cp-ansible-on-demand/2171/)
ARM64 + zookeeper -> [2160](https://jenkins.confluent.io/job/cp-ansible-on-demand/2060/) [2166](https://jenkins.confluent.io/job/cp-ansible-on-demand/2166/) 
ARM64 + kraft -> [2163](https://jenkins.confluent.io/job/cp-ansible-on-demand/2163/) [2167](https://jenkins.confluent.io/job/cp-ansible-on-demand/2167/)

# Checklist:

- [X] Any variable/code changes have been validated to be backwards compatible (doesn't break upgrade)
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If required, I have ensured the changes can be discovered by cp-ansible discovery codebase
- [X] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

[ANSIENG-2378]: https://confluentinc.atlassian.net/browse/ANSIENG-2378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/HOW_TO_TEST.md [hunk 1]
```diff
@@ -88,6 +88,45 @@ Simply run:
 molecule destroy -s <scenario name>
 ```
 
+### Running with some enviroment variables set
```
Comment (mansisinha) [inline docs/HOW_TO_TEST.md:91]: This is not needed. We can simply set env variables using export on the system and use Kraft/arm64 mode. Please remove this section from the docs. We use this passing of file system for automating the tests and decreasing manual work.

File: molecule/Dockerfile-centos8-base.j2 [hunk 2]
```diff
@@ -0,0 +1,41 @@
+FROM {{ item.image }}
```
Comment (mansisinha) [inline molecule/Dockerfile-centos8-base.j2:1]: why is this file needed? Is it not possible to use one of the already existing docker files?

File: molecule/Dockerfile-centos8-ldap.j2 [hunk 3]
```diff
@@ -0,0 +1,36 @@
+FROM {{ item.image }}
```
Comment (mansisinha) [inline molecule/Dockerfile-centos8-ldap.j2:1]: same. Why is this file needed?

File: molecule/confluent-kafka-kerberos-customcerts-rhel/molecule.yml [hunk 4]
```diff
@@ -11,8 +11,8 @@ platforms:
     hostname: kerberos1.confluent
     groups:
       - kerberos_server
-    image: geerlingguy/docker-centos7-ansible
-    dockerfile: ../Dockerfile.j2
+    image: centos:centos8
```
Comment (mansisinha) [inline molecule/confluent-kafka-kerberos-customcerts-rhel/molecule.yml:14]: why are we using centos:centos8 here instead of ubi8? does centos:centos8 have arm64 image?

File: molecule/rbac-mds-kerberos-debian/molecule.yml [hunk 5]
```diff
@@ -81,8 +81,8 @@ platforms:
     groups:
       - ${CONTROLLER_HOSTGROUP:-zookeeper}2
       - cluster2
-    image: geerlingguy/docker-debian9-ansible
-    dockerfile: ../Dockerfile-debian.j2
+    image: geerlingguy/docker-debian10-ansible
+    dockerfile: ../Dockerfile-debian10-archive.j2
```
Comment (mansisinha) [inline molecule/rbac-mds-kerberos-debian/molecule.yml:85]: why are we using Dockerfile-debian10-archive here? This is not an archive scenario

File: molecule/rbac-scram-custom-rhel-fips/molecule.yml [hunk 6]
```diff
@@ -16,8 +16,8 @@ platforms:
     hostname: ldap1${BUILD_NUMBER}.confluent${BUILD_NUMBER}
     groups:
       - ldap_server
-    image: geerlingguy/docker-centos7-ansible
-    # dockerfile: ../Dockerfile-rhel-java17.j2
+    image: centos:centos8
+    dockerfile: ../Dockerfile-centos8-ldap.j2
```
Comment (mansisinha) [inline molecule/rbac-scram-custom-rhel-fips/molecule.yml:20]: Do we need this dockerfile here? I believe this scenario does not use dockerfile on purpose to test some case

**Comment by mansisinha** [comment]
are we also testing centos7 and debian9 scenarios for arm64? What happens if I check arm64 on jenkins test on a centos7 scenario?

File: molecule/confluent-kafka-kerberos-customcerts-rhel/molecule.yml (hunk 4 above)
Comment (rrbadiani) [reply molecule/confluent-kafka-kerberos-customcerts-rhel/molecule.yml:14]: Both ubi8 and centos8 have amd and arm support. The difference is with the package `openldap-servers` In centos this package is available in the powertools repository vs in redhat/ubi8 it is paid.

File: molecule/rbac-scram-custom-rhel-fips/molecule.yml (hunk 6 above)
Comment (rrbadiani) [reply molecule/rbac-scram-custom-rhel-fips/molecule.yml:20]: In all rbac scenarios we need to install openldap-servers package which comes from the repository powertools. In geerlingguy/docker-centos7-ansible docker image it is readily available (i.e powertools repo is pre enabled) and hence we dont need any extra configuration in Dockerfile vs in centos8 docker image 2 additional things are required. 
- RUN yum -y config-manager --set-enabled powertools (to enable powertools repo which is by default in disabled state)
- Some commands to enable systemd (as this vanila centos8 image doesnt have systemd enabled by default)

File: test_roles/confluent.test.ldap/templates/certs.ldif.j2 [hunk 7]
```diff
@@ -8,17 +8,15 @@ olcTLSCACertificateFile: /var/ssl/private/ldaps/{{root_certificate_name}}
 olcTLSCACertificateFile: /var/ssl/private/ldaps/{{ssl_ca_cert_filepath|basename}}
 {%endif%}
 -
-delete: olcTLSCACertificatePath
--
-replace: olcTLSCertificateKeyFile
+add: olcTLSCertificateKeyFile
```
[the remark this answers is not in the document]
Comment (rrbadiani) [reply test_roles/confluent.test.ldap/templates/certs.ldif.j2:11]: Earlier this olcTLSCertificateKeyFile was getting created and put into some config file which we used to modify. But in this new centos8 image it wasnt created and put in the config file so replace was failing and hence I simply added it.

File: molecule/Dockerfile-centos8-ldap.j2 (hunk 3 above)
Comment (rrbadiani) [reply molecule/Dockerfile-centos8-ldap.j2:1]: Yes this one is not needed. Removing it

File: molecule/Dockerfile-centos8-base.j2 (hunk 2 above)
Comment (rrbadiani) [reply molecule/Dockerfile-centos8-base.j2:1]: This new Dockerfile is needed as we need to enable powertools repo using this file. Other Dockerfiles dont do that.

File: docs/HOW_TO_TEST.md (hunk 1 above)
Comment (rrbadiani) [reply docs/HOW_TO_TEST.md:91]: okay will remove this one

File: molecule/rbac-mds-kerberos-debian/molecule.yml (hunk 5 above)
Comment (rrbadiani) [reply molecule/rbac-mds-kerberos-debian/molecule.yml:85]: Actually `Dockerfile-debian10.j2` and `Dockerfile-debian10-archive.j2` only differ in one command. `Dockerfile-debian10-archive.j2` has extra command to `apt-key add & add-apt-repository`. So I didnt think it would matter whichever we would use. But to remain consistent with naming will change the Dockerfile in this scenario.

**Comment by rrbadiani** [comment]
> are we also testing centos7 and debian9 scenarios for arm64? What happens if I check arm64 on jenkins test on a centos7 scenario? now that we have updated OS for molecule.yml files, have we also checked and removed the docker files which are not getting used in any scenario now? Please update the base branch as master.

- Currently if we run tests on centos7 or debian9 they will fail as docker images used for those OS are not having arm64 architecture. 
- will check and remove extra Dockerfiles.
- updated the base branch to master

**Review by mansisinha — APPROVED** [review]
LGTM! Great Work!

**Review by mansisinha — APPROVED** [review]
